doc/classes/Node3D.xml

```diff
@@ -390,10 +390,10 @@
 			[b]Note:[/b] Some 3D nodes such as [CSGShape3D] or [CollisionShape3D] automatically enable this to function correctly.
 		</constant>
 		<constant name="ROTATION_EDIT_MODE_EULER" value="0" enum="RotationEditMode">
-			The rotation is edited using a [Vector3] in [url=https://en.wikipedia.org/wiki/Euler_angles]Euler angles[/url].
+			The rotation is edited using a [Vector3] in [url=https://en.wikipedia.org/wiki/Euler_angles]Euler angles[/url]. In Godot, Euler angles always use intrinsic order, meaning that rotation happens around the local axes of the object.
 		</constant>
 		<constant name="ROTATION_EDIT_MODE_QUATERNION" value="1" enum="RotationEditMode">
-			The rotation is edited using a [Quaternion].
+			The rotation is edited using a [Quaternion]. Quaternions avoid gimbal lock and having to choose an order of rotation, but are less intuitive. Quaternion rotation is mostly the same as rotors in 3D geometric algebra, except that the numbers are labeled differently.
```

Comment:
There's no explanation anywhere about what "gimbal lock" is, so this is (to your dismay) something that sounds needlessly complicated to the average user. You could briefly link an explanation to it or not mention this concept at all.

Reply:
Linking to an explanation makes sense, though to be blunt, part of the goal is to make Quaternion sound complicated. Most users don't need to click on this option, so it makes sense to me to put a bit of "scary" text on it.